src/lib/containers/CardContainerLogic.tsx

```diff
@@ -124,6 +124,25 @@ export default class CardContainerLogic extends React.Component<
     if (prevProps.sql !== this.props.sql) {
       this.executeInitialQueryRequest()
     }
+    const { searchParams: prevSearchParams = {} } = prevProps
+    const { searchParams: currentSearchParams = {} } = this.props
+    // check if search parameters have changed
+    if (
```

Comment:
Why not use 
[_isEqual()](https://lodash.com/docs/4.17.15#isEqual) 

May be something like this?
```
  if ((prevProps.sql !== this.props.sql) || (!_.isEqual(prevProps.searchParams,  this.props.searchParams))
      this.executeInitialQueryRequest()
    }
```
I think this is all that needs to be done. 
Another alternative would be to make searchParams immutable on the parent level that way a new object gets created every time it changes and you can just test the object equality. 

Also seems like it might be a good candidate for a fn component with a hook? (that was purely a question)

Reply:
I agree with making this a hook, I'd like to do that in another PR.